edge-detect/yolo/sort.py

```diff
@@ -0,0 +1,295 @@
+
+                    # Write results
+                    for *xyxy, conf, cls in reversed(det):
+                        if self.save_txt:  # Write to file
+                            xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
+                            line = (cls, *xywh, conf) if self.save_conf else (cls, *xywh)  # label format
+                            with open(txt_path + '.txt', 'a') as f:
+                                f.write(('%g ' * len(line)).rstrip() % line + '\n')
+
+                        if save_img or self.view_img:  # Add bbox to image
+                            label = f'{names[int(cls)]} {conf:.2f}'
+                            plot_one_box(xyxy, im0, label=label, color=colors[int(cls)], line_thickness=1)
+
+                # Print time (inference + NMS)
+                print(f'{s}Done. ({(1E3 * (t2 - t1)):.1f}ms) Inference, ({(1E3 * (t3 - t2)):.1f}ms) NMS')
+
+                # Stream results
+                if self.view_img:
+                    cv2.imshow(str(p), im0)
+                    cv2.waitKey(1)  # 1 millisecond
+
+                # Save results (image with detections)
+                if save_img:
+                    if dataset.mode == 'image':
+                        write_flag = cv2.imwrite(save_path, im0)
+                        if not write_flag:
+                            print('! Image write error')
+                        print(f" The image with the result is saved in: {save_path}")
+                    else:  # 'video' or 'stream'
+                        if vid_path != save_path:  # new video
+                            vid_path = save_path
+                            if isinstance(vid_writer, cv2.VideoWriter):
+                                vid_writer.release()  # release previous video writer
+                            if vid_cap:  # video
+                                fps = vid_cap.get(cv2.CAP_PROP_FPS)
+                                w = int(vid_cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+                                h = int(vid_cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+                            else:  # stream
+                                fps, w, h = 30, im0.shape[1], im0.shape[0]
+                                save_path += '.mp4'
+                            vid_writer = cv2.VideoWriter(save_path, cv2.VideoWriter_fourcc(*'mp4v'), fps, (w, h))
+                        vid_writer.write(im0)
+
+            # Move image file to destination folder based on model output
+            if pred[0].shape[0] > 0:
+                print('plane')
+                shutil.move(path, os.path.join(self.plane_dir, os.path.basename(path)))
+            else:
+                shutil.move(path, os.path.join(self.noplane_dir, os.path.basename(path)))
+                print('noplane')
+
+            if self.save_json:
+                struct = []
+                for rowidx in range(pred[0].shape[0]):
+                    row = pred[0][rowidx, :]
```

Comment:
@meadej  @dphogan - I think it makes sense to append some information about the model that ran the bounding box.  Can you add something in here with model type (yolo), and some kind of version of the model in use?  Not sure what specifically that looks like

Reply:
Addressed in https://github.com/IQTLabs/edgetech-yolo-detect/pull/3 and https://github.com/IQTLabs/edgetech-coral-detect/pull/1.